app/models/ebean/ReportingSchedule.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package models.ebean;
+
+import java.sql.Timestamp;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+/**
+ * Data Model for SQL storage of report schedules.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+@Entity
+@Table(name = "report_schedules", schema = "portal")
+// CHECKSTYLE.OFF: MemberNameCheck
+public class ReportingSchedule {
+    /**
+     * The recurrence behaviour of a schedule.
+     */
+    public enum RecurrenceType {
+        /**
+         * A non-recurring schedule.
+         */
+        NONE,
+        /**
+         * A schedule which recurs daily.
+         */
+        DAILY,
+        /**
+         * A schedule which recurs weekly.
+         */
+        WEEKLY
```

Comment:
So this union structure will capture all data for all types of schedules? e.g. third Monday of month, weekly from date x until date y, daily from date x for y occurrences, etc.

Reply:
Will this table also be updated with data necessary to evaluate the schedule? Or are we going to obtain that elsewhere? (e.g. most recently sent date)

Reply:
> So this union structure will capture all data for all types of schedules? e.g. third Monday of month, weekly from date x until date y, daily from date x for y occurrences, etc.

I updated this to be polymorphic in the type, from a quick run-through I think that should be sufficient to express any of the complex logic that we might want. (e.g. 3rd monday of the month, weekly on M W F)

> Will this table also be updated with data necessary to evaluate the schedule?

Job executions will be in a separate table, that data could be pulled out from there.